proxy/vm.go

```diff
@@ -270,6 +272,21 @@ func (session *ioSession) Close() {
 	session.wg.Wait()
 }
 
+func (vm *vm) CloseIo(seq uint64) {
+	vm.Lock()
+	session := vm.ioSessions[seq]
+	if session == nil {
+		vm.Unlock()
+		return
+	}
+	for i := 0; i < session.nStreams; i++ {
+		delete(vm.ioSessions, seq+uint64(i))
+	}
+	vm.Unlock()
+
+	session.Close()
```

Comment:
there's no returned error here :) Actually, the whole chunk is spurious and should be in that commit. Let me update that.